tests/MyEventEmitter.test.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const MyEventEmitter = require('../src/MyEventEmitter');
+const EventEmitter = require('events');
+
+describe('MyEventEmitter', () => {
+  let emitter;
+
+  beforeEach(() => {
+    emitter = new MyEventEmitter();
+  });
+
+  test('should not use external libraries', () => {
+    const sourceFile = path
+      .resolve(__dirname, '..', 'src', 'MyEventEmitter.js');
+    const fileContent = fs.readFileSync(sourceFile, 'utf8');
+    const hasCommonJSRequire = /require\(/g.test(fileContent);
+    const hasESImports = /import\s+\S+\s+from\s+['"].+['"]/g.test(fileContent);
+
+    expect(hasCommonJSRequire).toBe(false);
+    expect(hasESImports).toBe(false);
+  });
+
+  test('should not use Node.js EventEmitter', () => {
+    expect(emitter).not.toBeInstanceOf(EventEmitter);
+  });
+
+  test('should have working "listenerCount" method', () => {
+    expect(emitter.listenerCount('listenerCountEvent')).toBe(0);
+  });
+
+  test('should have working "on" method', () => {
+    const callback1 = jest.fn();
+    const callback2 = jest.fn();
+
+    expect(emitter.listenerCount('onEvent1')).toBe(0);
+
+    emitter.on('onEvent1', callback1);
+
+    expect(emitter.listenerCount('onEvent1')).toBe(1);
+
+    emitter.on('onEvent1', callback2);
+    emitter.on('onEvent2', callback2);
+
+    expect(emitter.listenerCount('onEvent1')).toBe(2);
+    expect(emitter.listenerCount('onEvent2')).toBe(1);
+  });
+
+  test('should have working "emit" method', () => {
```

Comment:
This name does not explain the details of the emit method
so please use more detailed explanations like

```
emit method should trigger callbacks added for a given event
```

Reply:
I don't think 1 test per method is enough